go-controller/pkg/ovn/egressip.go

```diff
@@ -2372,52 +2383,35 @@ func (oc *Controller) createDefaultNoRerouteNodePolicies(v4NodeAddr, v6NodeAddr
 	return nil
 }
 
-func (oc *Controller) createLogicalRouterPolicy(match string, priority int) error {
+func createLogicalRouterPolicy(nbClient libovsdbclient.Client, match string, priority int, externalIDs map[string]string) error {
 	lrp := nbdb.LogicalRouterPolicy{
-		Priority: priority,
-		Action:   nbdb.LogicalRouterPolicyActionAllow,
-		Match:    match,
+		Priority:    priority,
+		Action:      nbdb.LogicalRouterPolicyActionAllow,
+		Match:       match,
+		ExternalIDs: externalIDs,
 	}
 	p := func(item *nbdb.LogicalRouterPolicy) bool {
 		return item.Match == lrp.Match && item.Priority == lrp.Priority
 	}
-	err := libovsdbops.CreateOrUpdateLogicalRouterPolicyWithPredicate(oc.nbClient, types.OVNClusterRouter, &lrp, p)
+	err := libovsdbops.CreateOrUpdateLogicalRouterPolicyWithPredicate(nbClient, types.OVNClusterRouter, &lrp, p)
 	if err != nil {
 		return fmt.Errorf("error creating logical router policy %+v on router %s: %v", lrp, types.OVNClusterRouter, err)
 	}
 	return nil
 }
 
-func (oc *Controller) deleteLogicalRouterPolicy(match string, priority int) error {
+func DeleteDefaultNoRerouteNodePolicies(nbClient libovsdbclient.Client, node string) error {
 	p := func(item *nbdb.LogicalRouterPolicy) bool {
-		return item.Match == match && item.Priority == priority
-	}
-	err := libovsdbops.DeleteLogicalRouterPoliciesWithPredicate(oc.nbClient, types.OVNClusterRouter, p)
-	if err != nil {
-		return fmt.Errorf("error deleting router policy with priotity %d and match %s: %v", priority, match, err)
-	}
-
-	return nil
-}
-
-func (oc *Controller) deleteDefaultNoRerouteNodePolicies(v4NodeAddr, v6NodeAddr net.IP, v4ClusterSubnet, v6ClusterSubnet []*net.IPNet) error {
-	if v4NodeAddr != nil {
-		for _, v4Subnet := range v4ClusterSubnet {
-			match := fmt.Sprintf("ip4.src == %s && ip4.dst == %s/32", v4Subnet.String(), v4NodeAddr.String())
-			if err := oc.deleteLogicalRouterPolicy(match, types.DefaultNoRereoutePriority); err != nil {
-				return fmt.Errorf("unable to delete IPv4 no-reroute node policies, err: %v", err)
-			}
+		if item.Priority != types.DefaultNoRereoutePriority {
+			return false
 		}
-	}
-	if v6NodeAddr != nil {
-		for _, v6Subnet := range v6ClusterSubnet {
-			match := fmt.Sprintf("ip6.src == %s && ip6.dst == %s/128", v6Subnet.String(), v6NodeAddr.String())
-			if err := oc.deleteLogicalRouterPolicy(match, types.DefaultNoRereoutePriority); err != nil {
-				return fmt.Errorf("unable to delete IPv6 no-reroute node policies, err: %v", err)
-			}
+		nodeName, ok := item.ExternalIDs["node"]
```

Comment:
the legacy ones won't be updated but rather deleted by:
```
		if item.Priority != ovntypes.EgressSVCReroutePriority {
			return false
		}

		svcKey, found := item.ExternalIDs[svcExternalIDKey]
		if !found {
			klog.Infof("Egress service repair will delete lrp because it uses the egress service priority but does not belong to one: %v", item)
			return true
		}
```
as they use the priority that now belongs to egress services. then they will be recreated with the new priority (by the egressip controller or the new one). is that fine?

Reply:
yeah I think that is OK